### PR TITLE
Only change the sequence types if the target column type is a supported sequence type

### PR DESCRIPTION
**Pull request by onderkalaci**
DESCRIPTION: Fixes a bug that limits usage of sequences in non-int columns

Before this commit, we erroneously converted the sequence
type to the column's type it is used. However, it is possible
that the sequence is used in an expression which then converted
to a type that cannot be a sequence, such as text.

With this commit, we only try this conversion if the column
type is a supported sequence type (e.g., smallint, int and bigint).

Note that we do this conversion because if the column type is a
bigint and the sequence is NOT a bigint, users would be in trouble
because sequences would generate values that are out of the range
of the column. (The other ways are already not supported such as
the column is int and the sequence is bigint would fail on the worker.)

In other words, with this commit, we scope this optimization only
when the target column type is a supported sequence type. Otherwise,
we let users to more freely use the sequences.

Fixes https://github.com/citusdata/citus/issues/5759

### CODE REVIEW

File: src/backend/distributed/commands/create_distributed_table.c [hunk 1]
```diff
@@ -685,17 +686,29 @@ EnsureDistributedSequencesHaveOneType(Oid relationId, List *dependentSequenceLis
...
+		 * and int8 as the sequence type.
+		 *
 		 * A sequence's type is bigint by default and it doesn't change even if
 		 * it's used in an int column. We should change the type if needed,
 		 * and not allow future ALTER SEQUENCE ... TYPE ... commands for
-		 * sequences used as defaults in distributed tables
+		 * sequences used as defaults in distributed tables.
 		 */
-		AlterSequenceType(sequenceOid, seqTypId);
+		if (attributeTypeId == INT2OID ||
```
Comment (onderkalaci) [inline src/backend/distributed/commands/create_distributed_table.c:706]: We are still a little inconsistent if say the column is one these columns default is an expression using the sequence we change the type of sequence to bigint:
```C
c bigint DEFAULT nextval('int_sequence')  + 1000
``` 

For other types, we do not call `AlterSequenceType` when the sequence is used in an expression (e.g., otherwise  attributeTypeId would already be one of these types).

Think it is still preferable to follow this approach

**Review by marcocitus — APPROVED** [review]
